tests/integration/inverse-test.js

```diff
@@ -1,169 +1,392 @@
-import { run } from '@ember/runloop';
-import setupStore from 'dummy/tests/helpers/store';
-
-import testInDebug from 'dummy/tests/helpers/test-in-debug';
+import { setupTest } from 'ember-qunit';
 import { module, test } from 'qunit';
+import testInDebug from 'dummy/tests/helpers/test-in-debug';
+import Model from 'ember-data/model';
+import { attr, belongsTo, hasMany } from '@ember-decorators/data';
 
-import DS from 'ember-data';
-
-let env, store, User, Job, ReflexiveModel;
+module('integration/inverse-test - Model.inverseFor', function(hooks) {
+  setupTest(hooks);
+  let store;
 
-const { attr, belongsTo } = DS;
+  hooks.beforeEach(function() {
+    let { owner } = this;
 
-function stringify(string) {
-  return function() {
-    return string;
-  };
-}
+    class User extends Model {
+      @attr
+      name;
+      @belongsTo('user', { async: true, inverse: null })
+      bestFriend;
+      @belongsTo('job', { async: false })
+      job;
+    }
+    class Job extends Model {
+      @attr
+      isGood;
+      @belongsTo('user', { async: false })
+      user;
+    }
+    class ReflexiveModel extends Model {
+      @belongsTo('reflexiveModel', { async: false })
+      reflexiveProp;
+    }
+    owner.register('model:user', User);
+    owner.register('model:job', Job);
+    owner.register('model:reflexive-model', ReflexiveModel);
 
-module('integration/inverse_test - inverseFor', {
-  beforeEach() {
-    User = DS.Model.extend({
-      name: attr('string'),
-      bestFriend: belongsTo('user', { async: true, inverse: null }),
-      job: belongsTo('job', { async: false }),
-    });
+    store = owner.lookup('service:store');
+  });
 
-    User.toString = stringify('user');
+  test('Finds the inverse when there is only one possible available', function(assert) {
+    let Job = store.modelFor('job');
+    let inverseDefinition = Job.inverseFor('user', store);
 
-    Job = DS.Model.extend({
-      isGood: attr(),
-      user: belongsTo('user', { async: false }),
-    });
+    assert.deepEqual(
+      inverseDefinition,
+      {
+        type: store.modelFor('user'),
+        name: 'job',
+        kind: 'belongsTo',
+        options: {
+          async: false,
+        },
+      },
+      'Gets correct type, name and kind'
+    );
+  });
 
-    Job.toString = stringify('job');
+  test('Finds the inverse when only one side has defined it manually', function(assert) {
+    let { owner } = this;
+    owner.unregister('model:job');
+    owner.unregister('model:user');
 
-    ReflexiveModel = DS.Model.extend({
-      reflexiveProp: belongsTo('reflexive-model', { async: false }),
-    });
+    class Job extends Model {
+      @belongsTo('user', { inverse: 'previousJob', async: false })
```

Comment:
I think this is missing the user belongs to which was there before. The reopen just added an owner, it didn't clobber the user

Reply:
updated and restored and re-ordered the assertions to ensure we were capturing that.

I noted during the refactor that I was being "lossy" on this test from the original, but felt it didn't matter because the test was looking up the relationship via the `job` model first, which often avoids any issues of there being multiple potential inverses since it was the side with multiple.

I've updated the test to instead lookup the relationship via the `User` model first, and restored the extra `user` relationship on `Job`.